File: src/main/java/com/tfmg_energy_connector/ModEvents.java
package com.tfmg_energy_connector;

public class ModEvents {
}
package com.tfmg_energy_connector;

import net.neoforged.bus.api.SubscribeEvent;
import net.neoforged.fml.common.EventBusSubscriber;
import net.neoforged.neoforge.capabilities.Capabilities;
import net.neoforged.neoforge.capabilities.RegisterCapabilitiesEvent;

@EventBusSubscriber(modid = TfmgAe2Bridge.MODID, bus = EventBusSubscriber.Bus.MOD)
public class ModEvents {

    @SubscribeEvent
    public static void registerCapabilities(RegisterCapabilitiesEvent event) {
        // Говорим игре, что наш блок имеет энергию
        event.registerBlockEntity(
                Capabilities.Energy.BLOCK,
                TfmgAe2Bridge.CONVERTER_BE.get(),
                (blockEntity, context) -> blockEntity.getEnergyStorage()
        );
    }
}

File: src/main/java/com/tfmg_energy_connector/ModEvents.kt
package com.tfmg_energy_connector

import net.minecraft.world.item.CreativeModeTabs
import net.minecraft.world.item.Item
import net.neoforged.neoforge.event.BuildCreativeModeTabContentsEvent

object ModEvents {
    fun addCreative(event: BuildCreativeModeTabContentsEvent) {
        if (event.tabKey == CreativeModeTabs.FUNCTIONAL_BLOCKS) {
            val item: Item = TfmgAe2Bridge.CONVERTER_ITEM.get()
            event.accept(item)
        }
    }
}